analysis/cmd/worker.go

```diff
@@ -3,14 +3,28 @@ package main
 import (
 	"context"
 	"log"
+	"net/url"
 	"os"
+	"strings"
 
 	"gocloud.dev/pubsub"
 	_ "gocloud.dev/pubsub/gcppubsub"
 
 	"github.com/ossf/package-analysis/analysis"
 )
 
+func parseS3URL(s3URL string) string {
+	// This assumes OSSF_MALWARE_ANALYSIS_RESULTS given in the form s3://endpoint:port/bucket
+	// https://gocloud.dev/howto/blob/#s3-compatible
+
+	parsedURL, err := url.Parse(s3URL)
+	if err != nil {
+		log.Printf("s3 url for OSSF_MALWARE_ANALYSIS_RESULTS could not be parsed: %v", err)
+	}
+
+	return parsedURL.Scheme + ":/" + parsedURL.Path + "?endpoint=" + parsedURL.Host + "&disableSSL=true&s3ForcePathStyle=true"
```

Comment:
Can we just set `OSSF_MALWARE_ANALYSIS_RESULTS` in this format in the first place rather than trying to transform this?

Reply:
I think it's a little too much to expect the user to provide a formatted request url string. This could be avoided by using `s3blob.OpenBucket` directly when targeting an S3 endpoint but would require further handling to create the required AWS session.

Reply:
On second opinion I'll drop this PR, and assume the end user can provide a required string